examples/linear_model/plot_lasso_model_selection.py

```diff
@@ -1,171 +1,229 @@
 )
-plt.axvline(
-    model.alpha_ + EPSILON, linestyle="--", color="k", label="alpha: CV estimate"
-)
-
-plt.legend()
+plt.axvline(model[-1].alpha_, linestyle="--", color="black", label="alpha: CV estimate")
 
+plt.ylim(ymin, ymax)
 plt.xlabel(r"$\alpha$")
 plt.ylabel("Mean square error")
-plt.title(
-    "Mean square error on each fold: coordinate descent (train time: %.2fs)"
-    % t_lasso_cv
+plt.legend()
+_ = plt.title(
+    f"Mean square error on each fold: coordinate descent (train time: {fit_time:.2f}s)"
 )
-plt.axis("tight")
-plt.ylim(ymin, ymax)
 
-# #############################################################################
-# LassoLarsCV: least angle regression
+# %%
+# Lasso via least angle regression
+# ................................
+# Lets' start to make the hyperparameter tuning using
+# :class:`~sklearn.linear_model.LassoLarsCV`.
+from sklearn.linear_model import LassoLarsCV
 
-# Compute paths
-print("Computing regularization path using the Lars lasso...")
-t1 = time.time()
-model = LassoLarsCV(cv=20, normalize=False).fit(X, y)
-t_lasso_lars_cv = time.time() - t1
+start_time = time.time()
+model = make_pipeline(StandardScaler(), LassoLarsCV(cv=20, normalize=False)).fit(X, y)
+fit_time = time.time() - start_time
 
-# Display results
-plt.figure()
-plt.semilogx(model.cv_alphas_ + EPSILON, model.mse_path_, ":")
+# %%
+plt.semilogx(model[-1].cv_alphas_, model[-1].mse_path_, ":")
 plt.semilogx(
-    model.cv_alphas_ + EPSILON,
-    model.mse_path_.mean(axis=-1),
-    "k",
+    model[-1].cv_alphas_,
+    model[-1].mse_path_.mean(axis=-1),
+    color="black",
     label="Average across the folds",
     linewidth=2,
 )
-plt.axvline(model.alpha_, linestyle="--", color="k", label="alpha CV")
-plt.legend()
+plt.axvline(model[-1].alpha_, linestyle="--", color="black", label="alpha CV")
 
+plt.ylim(ymin, ymax)
 plt.xlabel(r"$\alpha$")
 plt.ylabel("Mean square error")
-plt.title("Mean square error on each fold: Lars (train time: %.2fs)" % t_lasso_lars_cv)
-plt.axis("tight")
-plt.ylim(ymin, ymax)
-
-plt.show()
+plt.legend()
+_ = plt.title(f"Mean square error on each fold: Lars (train time: {fit_time:.2f}s)")
+
+# %%
+# Summary of cross-validation approach
+# ....................................
+# Both algorithms give roughly the same results.
+#
+# Lars computes a path solution only for each kink in the path. As a result, it
+# is very efficient when there are only of few kinks, which is the case if
+# there are few features or samples. Also, it is able to compute the full path
+# without setting any meta parameter. On the opposite, coordinate descent
+# compute the path points on a pre-specified grid (here we use the default).
+# Thus it is more efficient if the number of grid points is smaller than the
+# number of kinks in the path. Such a strategy can be interesting if the number
+# of features is really large and there are enough samples to select a large
+# amount. In terms of numerical errors, for heavily correlated variables, Lars
```

Comment:
a large amount of what?

Reply:
Good catch this from the previous example. I assume when intended to say that when there are a lot of features to select from.

Reply:
Uhm not even. I don't know :)

Reply:
Since this is the CV-based lasso with coordinate descent, it could be that you need a sample size big enough in the CV fold